src/MailboxesParser/MailboxesParser.php

```diff
@@ -0,0 +1,245 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: Lukasz
+ * Date: 2017-11-10
+ * Time: 19:49
+ */
+
+namespace Ddeboer\Imap\MailboxesParser;
+
+
+use Ddeboer\Imap\Exception\MailboxesParserException;
+use Ddeboer\Imap\MailboxInterface;
+
+class MailboxesParser
+{
+    /** @var MailboxInterface[] */
+    protected $mailboxes;
+    /** @var ParsedMailbox[] */
+    protected $folders;
+    protected $treeStructure;
+
+    const INBOX = 'inbox';
+    const SENT = 'sent';
+    const DRAFT = 'drafts';
+    const SPAM = 'spam';
+    const TRASH = 'trash';
+    const TEMPLATES = 'templates';
+    const ARCHIVES = 'archives';
+
+    protected $specialFoldersIds = [
+        self::INBOX     => ['inbox',],
+        self::SENT      => ['sent', 'sent messages', 'INBOX.Sent', '[Gmail]/Sent Mail',],
+        self::DRAFT     => ['drafts', 'INBOX.Drafts', '[Gmail]/Drafts',],
+        self::SPAM      => ['spam', 'INBOX.spam', '[Gmail]/spam'],
+        self::TRASH     => ['trash', 'bin', 'INBOX.trash', '[Gmail]/trash'],
+        self::TEMPLATES => ['templates'],
+        self::ARCHIVES  => ['archives',],
+    ];
+
+    protected $specialFoldersNames = [
+        self::DRAFT     => 'Drafts',
+        self::INBOX     => 'Inbox',
+        self::SENT      => 'Sent',
+        self::SPAM      => 'Spam',
+        self::TRASH     => 'Trash',
+        self::TEMPLATES => 'Templates',
+        self::ARCHIVES  => 'Archives',
+    ];
```

Comment:
The same goes for folder names: if you need them, let the user specify them as a constructor dependency.

We don't have the resources to maintain translations.

Reply:
You can manually name special folders with `setSpecialFoldersNames`.
and add special folders Id with `addSpecialFolderId`